### ISSUE TITLE
is it possible to configure what metrics are being exposed?

### ISSUE THREAD

**Issue by f1-outsourcing** [zhangjianweibj/prometheus-libvirt-exporter#30]
I am not really interested in send the network info, nor tags like "projectName" etc. Can the exporter be configured what to send?

**Comment by frittentheke** [comment]
Dropping metrics can be achieved via the relabeling of Prometheus (https://www.robustperception.io/dropping-metrics-at-scrape-time-with-prometheus).

As for the labels you could remove / group on those at query time via PromQL. 

Another approach would be to move all those labels holding metadata to `_info` named metrics (https://www.robustperception.io/exposing-the-software-version-to-prometheus) which can be joined via a common key, i.e. the `instanceId` only when need be.

This technique is quite heavily used e.g. by kube-state-metrics (https://github.com/kubernetes/kube-state-metrics/blob/master/docs/node-metrics.md) exposing Kubernetes style resources, such as pods, nodes and their metadata / state info or the node_exporter (https://github.com/prometheus/node_exporter/blob/master/collector/diskstats_linux.go#L71).

but even though a best-practice and certainly a good way to have this exporter scale with more and more metadata fields being exported, this will be quite a change to the metrics and @zhangjianweibj would need to agree to follow this approach too.


https://github.com/prometheus/node_exporter/blob/9fbb56c9c80d5486d6de5e06a224a1eb343b28e7/collector/os_release.go#L77

**Comment by f1-outsourcing** [comment]
Yes but that is like: can you carry all the yellow and red bricks to the building site, and then at the building site use only the red bricks.
